frame/elections-phragmen/src/lib.rs

```diff
@@ -945,7 +969,13 @@ impl<T: Config> Pallet<T> {
 
 		let mut candidates_and_deposit = Self::candidates();
 		// add all the previous members and runners-up as candidates as well.
-		candidates_and_deposit.append(&mut Self::implicit_candidates_with_deposit());
+		match candidates_and_deposit
+			.try_append(&mut Self::implicit_candidates_with_deposit())
+			.map_err(|_| T::DbWeight::get().reads(3))
+		{
+			Ok(_) => (),
+			Err(weight) => return weight,
```

Comment:
Instead explicitly passing the weight when appending the candidates errors due to overloading the vec bounds, how about we return something like `T::WeightInfo::election_too_many_candidates()`? This requires a new benchmark for this error branch.

The benchmark approach is more future proof and easier to read/maintain.

Reply:
OTOH, I see the explicit weights are used a lot in `fn do_phragmen`, so I will leave up to you to decide which way to go.